nin/frontend/app/scripts/components/menubar.js

```diff
@@ -0,0 +1,306 @@
+            name: 'Toggle fullscreen',
+            action: 'toggleFullscreen',
+            defaultKeybind: 'enter',
+            invoke: () => commands.toggleFullscreen()
+          },
+          {
+            name: 'Mute',
+            action: 'toggleMute',
+            defaultKeybind: 'm',
+            invoke: () => commands.toggleMusic()
+          },
+          {
+            name: 'Play/pause',
+            action: 'togglePlayPause',
+            defaultKeybind: 'space',
+            charCode: '32',
+            invoke: () => commands.playPause()
+          }
+        ]
+      },
+      {
+        name: 'Render',
+        items: [
+          {
+            name: 'Start rendering',
+            action: 'startRendering',
+            invoke: () => commands.startRendering()
+          },
+          {
+            name: 'Stop rendering',
+            action: 'stopRendering',
+            invoke: () => commands.stopRendering()
+          }
+        ]
+      },
+      {
+        name: 'Generate',
+        items: [
+          {
+            name: 'Node',
+            action: 'generateNode',
+            invoke: () =>  {
+              commands.pause();
+              const nodeName = window.prompt("Enter a name for the node:");
+              commands.generate('node', nodeName);
+            }
+          },
+          {
+            name: 'THREE Node',
+            action: 'generateTHREENode',
+            invoke: () =>  {
+              commands.pause();
+              const nodeName = window.prompt("Enter a name for the node:");
+              commands.generate('threeNode', nodeName);
+            }
+          },
+          {
+            name: 'Shader Node',
+            action: 'generateShaderNode',
+            invoke: () =>  {
+              commands.pause();
+              const nodeName = window.prompt("Enter the shader's name in CamelCase, like 'SunShader':");
+              commands.generate('shaderNode', nodeName);
+            }
+          }
+        ]
+      },
+      {
+        name: 'Theme',
+        items: [
+          {
+            name: 'Dark',
+            action: 'activateDarkTheme',
+            invoke: () => commands.selectTheme('dark')
+          },
+          {
+            name: 'Light',
+            action: 'activateLightTheme',
+            invoke: () => commands.selectTheme('light')
+          }
+        ]
+      },
+      {
+        name: 'Help',
+        items: [
+          {
+            name: 'Online wiki',
+            action: 'help',
+            invoke: () => $window.open('https://github.com/ninjadev/nin/wiki')
+          }
+        ]
+      }
+    ];
+
+    this.menu.forEach(category => {
+      category.items.forEach(item => {
+        item.keybind =
+          ninrc.keybinds && ninrc.keybinds[item.action]
+          || item.defaultKeybind;
+        if (item.keybind) {
+          keymage(item.keybind, item.invoke);
+        }
+      });
+    });
+
+    this.state = {
+      open: false,
+      focus: ''
```

Comment:
What exactly did oldTopMenuOpen do? I couldn't figure out how it was needed, so I just removed it.